Jenkinsfile

```diff
@@ -415,12 +415,28 @@ for(String envKeyName: context.env.keySet() as String[]){
                     //    sh 'python manage.py loaddata wells registries'
                     //}
                     dir('functional-tests') {
+                        Integer attempts = 0
+                        Integer attemptsMax = 2
                         try {
-                            //sh './gradlew -q dependencies'
-                            if ("DEV".equalsIgnoreCase(stageDeployName)){
-                                sh './gradlew chromeHeadlessTest'
-                            }else{
-                                sh './gradlew -DchromeHeadlessTest.single=WellDetails chromeHeadlessTest'
+                            waitUntil {
+                                boolean isDone=false
+                                attempts++
+                                try{
+                                    if ("DEV".equalsIgnoreCase(stageDeployName)){
+                                        sh './gradlew chromeHeadlessTest'
+                                    }else{
+                                        sh './gradlew -DchromeHeadlessTest.single=WellDetails chromeHeadlessTest'
+                                    }
+                                    isDone=true
+                                } catch (ex) {
+                                    echo "${stackTraceAsString(ex)}"
+                                    echo "DEV - Functional Tests Failed - Wait one minute and retry once"
+                                    sleep 60
+                                }
+                                if ( attempts >= attemptsMax ){
```

Comment:
I think we may need to move this if block inside the catch block above, and re-throw the exception, otherwise it will just fail silently.
``if ( attempts >= attemptsMax ){
throw ex
}
``

Reply:
Done!  Good catch.